pkgs/test_api/lib/src/backend/compiler.dart

```diff
@@ -0,0 +1,51 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// All the Dart compilers supported by the test runner.
+class Compiler {
+  /// The production Dart to Javascript compiler (whole world, optimizing).
+  static const Compiler dart2js =
+      Compiler('Dart2Js', 'dart2js', isDefault: true);
+
+  /// Experimental Dart to Wasm compiler.
+  static const Compiler dart2wasm = Compiler('Dart2WASM', 'dart2wasm');
+
+  /// The standard compiler for vm tests, compiles tests to kernel before
+  /// running them on the VM.
+  static const Compiler kernel = Compiler('Kernel', 'kernel', isDefault: true);
+
+  /// Runs tests directly from source, with no precompilation.
+  static const Compiler source = Compiler('Source', 'source');
+
+  /// The compilers that are supported by the test runner by default.
+  static const List<Compiler> builtIn = [
+    Compiler.dart2js,
+    Compiler.dart2wasm,
+    Compiler.kernel,
+    Compiler.source,
+  ];
+
+  /// The human-friendly name of the compiler.
+  final String name;
+
+  /// The identifier used to look up the compiler.
+  final String identifier;
+
+  /// Whether this compiler is enabled by default.
+  final bool isDefault;
```

Comment:
I'm still skimming the change, but I find it confusing that being default is a property of a Compiler, and not a property of a Platform/Compiler relationship.

I could imagine that `dart2js` might remain a default on node, but ddc could become a default for the browser.

Reply:
Good catch this is definitely just cruft - I need to remove it.